### DISCUSSION TITLE
Log: testing.ERROR: Disk [tmp-for-tests] does not have a configured driver.

### DISCUSSION THREAD

**Discussion by CamKem** [livewire/livewire#8667] [1. Help] [answered]
I have noticed in the logs, when working on unit tests an image upload feature and any unit test where I set the value of a property to a fake UploadedFiled is triggering an error in the logs for: 

[2024-07-18 08:06:47] testing.ERROR: Disk [tmp-for-tests] does not have a configured driver. {"userId":1,"exception":"[object] (InvalidArgumentException(code: 0): Disk [tmp-for-tests] does not have a configured driver. at /Users/cam/Sites/pinkary-repo/vendor/laravel/framework/src/Illuminate/Filesystem/FilesystemManager.php:137)

I posted this in a discussion, however, it seems this is not the same issue they were having.
Any assistance would be appreciated.

This is an example test & I can provide the full stack trace if needed.
```php
test('used images are NOT cleanup when store is called', function () {
    Storage::fake('public');
    $user = User::factory()->create();
    $file = UploadedFile::fake()->image('photo1.jpg');
    $name = $file->hashName();
    $date = now()->format('Y-m-d');
    $path = 'images/'.$date.'/'.$name;

    $component = Livewire::actingAs($user)->test(Create::class, [
        'toId' => $user->id,
    ]);
    $component->set('images', [$file]);

    Storage::disk('public')->assertExists($path);

    expect(session('images'))->toBeArray()
        ->and(session('images'))->toContain($path);

    $url = Storage::disk('public')->url($path);

    $component->set('content', "![Image Alt Text]({$url})");
    $component->call('store');

    Storage::disk('public')->assertExists($path);
    expect(session('images'))->toBeNull();
});
```

_Originally posted by @CamKem in https://github.com/livewire/livewire/discussions/3105#discussioncomment-10081620_

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

**Reply by CamKem** [reply, marked as answer]
Hi Josh,

I haven't been able to figure out why this is causing log errors each time the tests run, however, if I add the driver to the filesystem.php, it will prevent the log errors from appearing each time the tests run. It was my understanding from looking at FileUploadConfiguration class that this shouldn't be required, it also poses a problem for apps which delete the filesystem.php config file to default back to the laravel framework built-in config file.

```php
    'disks' => [

        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
            'throw' => false,
        ],

        'tmp-for-tests' => [
            'driver' => 'local',
            'root' => storage_path('framework/testing/disks/tmp-for-tests'),
            'throw' => false,
        ],
```

Do you, or anyone else have any further thoughts on this?
